keep long running obd network tests alive

wlan0 opened the pull request:
@harshavardhana - in case of really large clusters, or clusters with inconsistent latencies, we need to keep the connection alive.

Tested it in our DC

wlan0 committed:
keep long running obd network tests alive

wlan0 committed:
fix error - wrong number of parents in process OBD info

wlan0 committed:
ensure that osinfo does not error out when inside containers

klauspost commented on cmd/peer-rest-server.go:
```diff
@@ -403,6 +403,9 @@ func (s *peerRESTServer) DispatchNetOBDInfoHandler(w http.ResponseWriter, r *htt
 	}
 
+	done := keepHTTPResponseAlive(w)
+	defer done()
```
You must call done before you write to the connection.

harshavardhana replied:
@wlan0 ^^

wlan0 replied:
done()

klauspost reviewed (dismissed):
Rest LGTM.

wlan0 committed:
Remove limit on max number of connections per client transport

The generic client transport uses a default limit of 64 conns per transport.
This could end up limiting and throttling usage, and artificially slowing
down the performance of MinIO even on hardware capable of doing better.
